src/Commands/BuildCommand.php

```diff
@@ -7,37 +7,59 @@
 use Illuminate\Support\Str;
 use Native\Electron\Concerns\LocatesPhpBinary;
 use Native\Electron\Facades\Updater;
-
-use function Laravel\Prompts\select;
+use Native\Electron\Traits\OsAndArch;
 
 class BuildCommand extends Command
 {
     use LocatesPhpBinary;
+    use OsAndArch;
+
+    protected $signature = 'native:build
+        {os? : The operating system to build for (all, linux, mac, win)}
+        {arch? : The Processor Architecture to build for (x64, x86, arm64)}
+        {--publish : to publish the app}';
 
-    protected $signature = 'native:build {os? : The operating system to build for (all, linux, mac, win)}';
+    protected $availOs = ['win', 'linux', 'mac', 'all'];
 
     public function handle(): void
     {
         $this->info('Build NativePHP app…');
 
+        Process::path(__DIR__.'/../../resources/js/')
+            ->env($this->getEnvironmentVariables())
+            ->forever()
+            ->run('npm update', function (string $type, string $output) {
+                echo $output;
+            });
+
         Process::path(base_path())
             ->run('composer install --no-dev', function (string $type, string $output) {
                 echo $output;
             });
 
-        if (! $os = $this->argument('os')) {
-            $os = select(
-                label: 'Please select the operating system to build for',
-                options: ['win', 'linux', 'mac', 'all'],
-                default: $this->getDefaultOs(),
-            );
+        // Added checks for correct input for os and arch
+        $os = $this->selectOs($this->argument('os'));
+
+        $buildCommand = 'build';
+        if ($os != 'all') {
+            $arch = $this->selectArchForOs($os, $this->argument('arch'));
+
+            $os .= $arch != 'all' ? "-{$arch}": '';
+
+            // Wether to publish the app or not
+            if ($publish = ($this->option('publish'))){
+                $buildCommand = 'publish';
+            }
         }
+        $this->info((($publish ?? false) ? "Publishing" : 'Building') . " for {$os}");
+
+        dd("npm run {$buildCommand}:{$os}");
```

Comment:
Did you meant to leave this `dd` in ?

Reply:
thats something I've missed. Probably because I was just checking for the right command, rather then building it and then forgot to remove it.